### PR TITLE
use more nameof keyword for generate code

### PR DESCRIPTION
**Pull request by Meir017**
This will cleanup the file [`Syntax.xml.Main.Generated.cs`](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Generated/Syntax.xml.Main.Generated.cs#L4301) and use `nameof` instead of constant strings

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Can you regenerate the files as well?

File: src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs [hunk 1]
```diff
@@ -1747,7 +1747,7 @@ private void WriteRedFactory(Node nd)
                 }
                 WriteLine("          break;");
                 WriteLine("        default:");
-                WriteLine("          throw new ArgumentException(\"kind\");");
+                WriteLine("          throw new ArgumentException(nameof(kind();");
```
Comment (sharwell) [inline src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs]: ❕ 

```suggestion
                 WriteLine("          throw new ArgumentException(nameof(kind));");
```

Comment (Meir017) [reply src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs]: oops

**Comment by CyrusNajmabadi** [comment]
@Meir017 If you want to clean up some more, this file could definitely benefit from using `=>` for simple methods/properties :)

**Comment by Meir017** [comment]
I hope it's not too many changes 😄

**Comment by jcouv** [comment]
>         public override TreeDumperNode VisitFieldInfo(BoundFieldInfo node, object arg) => new TreeDumperNode("fieldInfo", null, 

This seems slightly worse than before in terms of formatting.
One way to fix would be to indent the `new TreeDumperNode[] { ... }`.
Another would be to bring the `new TreeDumperNew[]` to the previous line.

---
Refers to: src/Compilers/CSharp/Portable/Generated/BoundNodes.xml.Generated.cs:8726 in eb93e48. [](commit_id = eb93e48c49336f80230a74e4e080309e19e900c5, deletion_comment = False)

**Comment by jcouv** [comment]
>         private static Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.IdentifierNameSyntax GenerateIdentifierName() => Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.SyntaxFactory.IdentifierName(Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.SyntaxFactory.Identifier("Identifier"));

This is also worse than before, because of extra long lines.

---
Refers to: src/Compilers/CSharp/Test/Syntax/Generated/Syntax.Test.xml.Generated.cs:12 in eb93e48. [](commit_id = eb93e48c49336f80230a74e4e080309e19e900c5, deletion_comment = False)

**Comment by CyrusNajmabadi** [comment]
Sorry, i should have been clearer: i woudl do the other cleanup in other PRs.

**Comment by CyrusNajmabadi** [comment]
If you do continue with this, consider the IDE style where `=>` is wrapped to the next line and indented (in the case where things are long).  Where things are short, seems fine to have `=>` continue to the right of the construct.

**Comment by CyrusNajmabadi** [comment]
> Generally, we do not take changes that focus on style, but this one is a change to generator, so may be ok.

Note: as someone who often has to f12 into the syntax-apis, this would be very appreciated.  Right now there is so much excess cruft and space that figuring out what you need from that file is pretty painful.  So, in this sense, while it is just "style" it's style that goes a *long* way to helping out anyone who needs to work with this file.

**Comment by Meir017** [comment]
I think I'm done 😄

**Comment by jcouv** [comment]
I'm waiting for CI to complete, so this can be merged.

**Comment by jcouv** [comment]
Merged. Thanks

**Pull request merged**